megatron/model/encoders/vision/dinov2/layers/block.py

```diff
@@ -74,13 +74,16 @@ def __init__(
 
         self.norm2 = norm_layer(dim)
         mlp_hidden_dim = int(dim * mlp_ratio)
-        self.mlp = ffn_layer(
-            in_features=dim,
-            hidden_features=mlp_hidden_dim,
-            act_layer=act_layer,
-            drop=drop,
-            bias=ffn_bias,
-        )
+        if ffn_layer == Mlp:
+            self.mlp = ffn_layer(
+                in_features=dim,
+                hidden_features=mlp_hidden_dim,
+                act_layer=act_layer,
+                drop=drop,
+                bias=ffn_bias,
+            )
+        elif ffn_layer == nn.Linear:
+            self.mlp = ffn_layer(dim, dim, bias=ffn_bias)
```

Comment:
I don't really understand why this was needed? When are we creating the mlp as a linear only?